hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -55,6 +55,12 @@ abstract class TableStage(
       broadcastVals.foldLeft(partition(ctx))((accum, name) => Let(name, GetField(glob, name), accum)))
     if (bind) wrapInBindings(cda) else cda
   }
+
+  def castPartitioner(newPartitioner: RVDPartitioner): TableStage = {
```

Comment:
This is not a cast.  It is just a new partitioner.  `with[New]Partitioner`?  `setPartitioner`?  `copy`?